sdk/spring/azure-spring-integration-servicebus/src/test/java/com/azure/spring/integration/servicebus/ServiceBusMessageHandlerTest.java

```diff
@@ -33,4 +58,99 @@ public void setUp() {
             .thenReturn(future);
         this.handler = new DefaultMessageHandler(this.destination, this.sendOperation);
     }
+
+    /**
+     * This is only for creating PartitionSupply setting
+     */
+    private void setPropertiesForPartitionSupplier() {
+        partitionIdExpression = EXPRESSION_PARSER.parseExpression("headers['scst_partition']");
+        partitionKeyExpression = new FunctionExpression<Message<?>>(m -> m.getPayload().hashCode());
+        payloadBytes = payload.getBytes(StandardCharsets.UTF_8);
+
+        this.handler.setBeanFactory(beanFactory);
+        this.handler.afterPropertiesSet();
+    }
+
+    @Test
+    public void testToPartitionSupplierReturnPartitionIdFromHeader() {
+        setPropertiesForPartitionSupplier();
+
+        // set partition id with integer type
+        Message<?> message = new GenericMessage<>(payloadBytes, ImmutableMap.of(AzureHeaders.PARTITION_ID, 1));
+        PartitionSupplier partitionSupplier = ReflectionTestUtils.invokeMethod(this.handler,
+            DefaultMessageHandler.class, TO_PARTITION_SUPPLIER_METHOD_NAME, message);
```

Comment:
Can we just build `PartitionSupplier` and test it without using `ReflectionTestUtils`?